internal/backup.go

```diff
@@ -85,21 +116,23 @@ func (backup *Backup) FetchMetadata(metadataDto interface{}) error {
 	return errors.Wrap(err, "failed to unmarshal metadata")
 }
 
-func (backup *Backup) fetchStorageBytes(path string) ([]byte, error) {
+func (backup *Backup) fetchStorageStream(path string) (io.ReadCloser, error) {
 	backupReaderMaker := NewStorageReaderMaker(backup.Folder, path)
-	backupReader, err := backupReaderMaker.Reader()
-	if err != nil {
-		return make([]byte, 0), err
-	}
-	metadata, err := ioutil.ReadAll(backupReader)
-	if err != nil {
-		return nil, err
-	}
-	return metadata, nil
+	return backupReaderMaker.Reader()
 }
 
 func (backup *Backup) UploadMetadata(metadataDto interface{}) error {
 	metaFilePath := backup.getMetadataPath()
+	if CommonAllowedSettings[UseSerializedJSONSetting] {
```

Comment:
This condition is always false. Look how we get the other settings, i.e. `viper.GetBool(internal.UseWalDeltaSetting)`